src/mantine-form/src/use-form.ts

```diff
@@ -50,6 +54,8 @@ export function useForm<
   const [dirty, setDirty] = useState(initialDirty);
   const [values, _setValues] = useState(initialValues);
   const [errors, _setErrors] = useState(filterErrors(initialErrors));
+  const [submitting, _setSubmitting] = useState(false);
+  const [submitError, _setSubmitError] = useState<SubmitError>(null);
```

Comment:
I think these should get reset in the [`reset` callback](https://github.com/mantinedev/mantine/pull/4020/files#diff-885f0cbfb62b24bda2ed7145a69597b51a67fabc68b78fde5b672962ed97559eR79)